docs/feature_combo.md

```diff
@@ -71,26 +92,118 @@ void process_combo_event(uint16_t combo_index, bool pressed) {
 
 |Keycode   |Description                      |
 |----------|---------------------------------|
 |`CMB_ON`  |Turns on Combo feature           |
 |`CMB_OFF` |Turns off Combo feature          |
 |`CMB_TOG` |Toggles Combo feature on and off |
 
+## Combo Term
+By default, the timeout for the Combos to be recognized is set to 50ms. This can be changed if accidental combo misfires are happening or if you're having difficulties pressing keys at the same time. For instance `#define COMBO_TERM 40` would set the time out period for combos to 40ms.
+
+## Long Combos
+If you're using long combos, or you have a lot of overlapping combos, you may run into issues with this, as the structure may not be large enough to accommodate what you're doing. In this case, you can configure the size of the data structure used. Be aware, larger combo sizes will increase memory usage!
+
+| Keys | Define to be set     |
+|------|----------------------|
+| 8    | QMK Default          |
+| 16   | `#define EXTRA_LONG_COMBOS`  |
+| 32   | `#define EXTRA_EXTRA_LONG_COMBOS`|
+
+## Modifier Combos
+If a combo resolves to a Modifier, the window for processing the combo can be extended independent from normal combos. By default this is disabled but can be enabled with `#define COMBO_MUST_HOLD_MODS`, and the time window can be configured with `#define COMBO_MOD_TERM 150` (default: 200). With `COMBO_MUST_HOLD_MODS` you cannot tap the combo any more which makes the combo less prone to misfires.
+
+## Per Combo Timing and Holding
+Instead of using a blanket `COMBO_TERM` window for every combo, per-combo timings can be set. In order to use this feature the following config options and functions need to be defined. Coming up with useful timings and configuration is left as an exercise for the reader.
+
+| Config Flag                 | Function                                                  | Description                                                                                          |
+|-----------------------------|-----------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `COMBO_MUST_HOLD_PER_COMBO` | bool get_combo_must_hold(uint16_t index, combo_t \*combo) | Controls if a given combo should fire immediately on tap or if it needs to be held. (default: False) |
+| `COMBO_TERM_PER_COMBO`      | uint16_t get_combo_term(uint16_t index, combo_t \*combo)  | Optional per-combo timeout window. (default: COMBO_TERM)                                             |
+
+If you set either of these options but would like to keep the Modifier Combo settings, you have to define those yourself in the function.
+
+Examples:
+```c
+uint16_t get_combo_term(uint16_t index, combo_t *combo) {
+    // decide by combo->keycode
+    if (KEYCODE_IS_MOD(combo->keycode)) return COMBO_MOD_TERM; // you have to config this yourself if you're using COMBO_TERM_PER_COMBO
```

Comment:
```suggestion
    if (KEYCODE_IS_MOD(combo->keycode)) {
        // you have to config this yourself if you're using COMBO_TERM_PER_COMBO
        return COMBO_MOD_TERM;
    }
```
Putting it all on one line makes it too long and forces (most of) the reader(s) to scroll horizontally to be able to read it all.